### PR TITLE
Feature/cvsb 5312

### PR DESCRIPTION
**Pull request by KHodderDVSA**
Add logging to all exception catching locations, and as much x-ray capture as I could work out how to correctly implement.
Replaced some console.error with console.log as cloudwatch wasn't showing console.error messages

### CODE REVIEW

File: src/functions/getTestResultsByTesterStaffId.js [hunk 1]
```diff
@@ -15,17 +22,25 @@ const getTestResultsByTesterStaffId = (event) => {
 
   if (!event.queryStringParameters) {
     if (event.queryStringParameters.testerStaffId && event.queryStringParameters.testStationPNumber && event.queryStringParameters.toDateTime && event.queryStringParameters.fromDateTime) {
+      console.log('Bad request in getTestResultsByTesterStaffId - missing required parameters');
```
Comment (acozianu) [inline src/functions/getTestResultsByTesterStaffId.js]: Please move the string in the enum file

File: src/functions/getTestResultsByVin.js [hunk 2]
```diff
@@ -14,24 +22,35 @@ const getTestResultsByVin = (event) => {
...
-      return Promise.resolve(new HTTPResponse(400, 'Bad Request'))
-    } else {
-      if (event.queryStringParameters.status) { testStatus = event.queryStringParameters.status }
-      if (event.queryStringParameters.toDateTime) { toDateTime = new Date(event.queryStringParameters.toDateTime) }
-      if (event.queryStringParameters.fromDateTime) { fromDateTime = new Date(event.queryStringParameters.fromDateTime) }
+  try {
+    if (event.queryStringParameters) {
+      if (event.queryStringParameters.toDateTime === '') {
+        if (subseg) {subseg.addError('Bad Request - toDate empty');}
+        console.log('Bad Request in getTestResultsByVin - toDate empty');
```
Comment (acozianu) [inline src/functions/getTestResultsByVin.js]: same here

File: src/functions/getTestResultsByTesterStaffId.js (hunk 1 above)
Comment (KHodderDVSA) [reply src/functions/getTestResultsByTesterStaffId.js]: Is there any point using enums for unique strings? They're only used the once in that specific circumstance

File: src/functions/getTestResultsByVin.js (hunk 2 above)
Comment (KHodderDVSA) [reply src/functions/getTestResultsByVin.js]: as above

File: src/functions/getTestResultsByTesterStaffId.js [hunk 3]
```diff
@@ -15,17 +22,25 @@ const getTestResultsByTesterStaffId = (event) => {
...
+      if (subseg) subseg.addError('Bad request in getTestResultsByTesterStaffId - missing required parameters');
```
Comment (bsiddappa) [inline src/functions/getTestResultsByTesterStaffId.js]: Can we have this if statement block inside try block and such that subseg gets closed in finally block?

File: src/functions/postTestResults.js [hunk 4]
```diff
@@ -1,26 +1,43 @@
...
+    subseg = segment.addNewSubsegment('postTestResults');
+  }
   const testResultsDAO = new TestResultsDAO()
   const testResultsService = new TestResultsService(testResultsDAO)
 
   let payload = event.body
 
   if (!payload) {
-    return Promise.resolve(new HTTPResponse(400, 'Body is not a valid JSON.'))
+    if (subseg) { subseg.addError(MESSAGES.INVALID_JSON); }
```
Comment (bsiddappa) [inline src/functions/postTestResults.js]: same holds here too.

File: src/functions/getTestResultsByTesterStaffId.js (hunk 3 above)
Comment (KHodderDVSA) [reply src/functions/getTestResultsByTesterStaffId.js]: fixed

File: src/functions/postTestResults.js (hunk 4 above)
Comment (KHodderDVSA) [reply src/functions/postTestResults.js]: fixed

**Pull request merged**